doc/SEMGREP_CORE_CONTRIBUTING.md

```diff
@@ -0,0 +1,535 @@
+
+Once you have changed `pfff`, `semgrep-core` will realize that `pfff` is different. If you go back up to `semgrep-core/` and run `git status`, you will see `modified: src/pfff (modified content)`. To pin your latest `pfff` changes to `semgrep-core`, add `src/pfff`.
+
+Now, make the rest of your pull request for `semgrep-core` as usual.
+
+If you haven't changed `pfff`, don't worry about this. Just make a pull request with your changes.
+
+Remember to add test cases so that future changes don't break your example! See [Testing `semgrep-core`](#testing-semgrep-core)
+
+### Fixing a Tree-sitter Parse Error
+
+There is more information in [Adding Support for a Language](#adding-support-for-a-language) on tree-sitter which will be helpful. Also, see `semgrep-core/src/parsing/tree-sitter/`.  
+
+## Fixing a Match Error
+
+The first thing you will need to do is understand what you expected and why you aren't getting that. If possible, reduce your rule to a single pattern that doesn't match. You may need to experiment with the clauses in your rule. For example, if you are getting too many matches, it may be because the pattern in `pattern-not` doesn't match what you expect.
+
+If you are unable to do so, you may need to investigate `Match_rule.ml`.
+
+Otherwise, produce a minimal failing pattern/target pair. You will need to compare the ASTs to see which portion is not matching as you expect. Run
+
+```
+sc -dump_ast [your_target].py -lang py
+```
+
+and then
+
+```
+sc -dump_pattern [your_pattern].sgrep -lang py
+```
+
+It can be hard to figure out where in the AST you are looking. You can make it easier by using a distinctive variable name in the section you're interested in. 
+
+Once you've isolated the parts that aren't matching, try to figure out where they're different, taking into account special features like metavariables and ellipses. It is unlikely (though not impossible) that the problem would ever be that two identical code segments aren't matching or that there is some AST element that ellipses refuse to match. You might find it helpful to write out the AST parts you want to match on a whiteboard, indicating which part is matched by a special feature. Pare down the code as much as possible and try changing the bit you're interested in.
+
+When you are sure you know what ought to have happened, make it happen. If two pieces of code should match but don't, change `Generic_vs_generic.ml` to tell it that pattern should match the target.
+
+Oftentimes, a matching error is actually a parsing error. You may want to change how `Parser_python.mly` reduces the construct or how it gets converted in `Python_to_generic.ml`. Refer to [Fixing a Parse Error](#fixing-a-parse-error) for advice.
+
+At the end, confirm the match with
+
+```
+sc -f [your_pattern].sgrep [your_target].py -lang py
+```
```

Comment:
Maybe you can say that the playground has also the debugging options to dump the ASTs of the pattern and target.